pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -1015,42 +1015,41 @@ struct ArgSortOpConversion : public OpConversionPattern<tile::ArgSortOp> {
       minVal = rewriter.create<LoadOp>(loc, minValVar, zeroIndex).getResult();
       // Is compVal smaller than minVal? If so, update the allocs
       Value orderPred;
-      if (elementType.isSignedInteger()) {
```

Comment:
Why are these cases being reordered?

Reply:
Using the test case below:
```
TEST_F(CppEdsl, ArgSort1dInt) {
  auto I = Placeholder(DType::INT32, {10});
  auto O = argsort(I, /*axis=*/0);
  auto program = makeProgram("argsort", {I}, {O});
  std::vector<int32_t> input = {
    0, 1, 8, 5, 5, 2, 0, 7, 7, 10,
  }
  std::vector<int32_t> output = {
    0, 6, 1, 5, 3, 4, 7, 8, 2, 9,
  }
  checkExact(program, {input}, {output});
```
Will raise an error:
```
error: 'std.cmpf' op operand #0 must be floating-point-like, but got 'i32'
  note: see current operation: %13 = "std.cmpf"(%11, %12) {predicate = 4 : i64} : (i32, i32) -> i1
```
Integer input tensor is not tested and it seems this will run into the "Assume float" case which causes type mismatch. I guess `elementType.isSignlessInteger()` is missing so I simply follow what `convertAgg` function at line 552 does. In short:
```
if (type.isa<FloatType>()) {
  ...
} else if (type.isSignedInteger()) {
  ...
} else {
  ...
}
```
And the error's gone. Please let me know if there's anything else missing.

Reply:
I think it makes sense for these cases and the cases in line 552 to match. So I think this is a good format for this PR.

If we want to make adjustments, I propose we make a separate PR and change both of these sections (and any other similar sections) in that PR.

Reply:
This follows other code that also falls back to unsigned if it's not float or signed. Just FYI, we don't use signless integer types at the edsl/tile level. 

Also, nit: I'd try to break up this function, it's kind of long. Maybe make a helper function that just returns the comparison op (like something that returns the value for `orderPred`).

Reply:
The test case above falls back to the last case (presumably unsigned) but before this change, there was a case that explicitly used `else if (type.isUnsignedInteger())` and it didn't match the test. This is why I guess it's a `SignlessInteger` case. If it's not, I'm not sure why it falls back to the last case then.

As for making a helper function, I updated the code and please check.

Reply:
I'm curious why we're seeing a signless integer in this case, but it's not a blocker for this PR

Reply:
I will try to figure it out.